src/main/java/com/siemens/ct/exi/main/api/dom/DOMWriter.java

```diff
@@ -124,7 +130,7 @@ public void encode(Node n) throws EXIException, IOException {
 		}
 	}
 
-	protected void encodeNode(Node root) throws EXIException, IOException {
+	protected void encodeNode(final Node root) throws EXIException, IOException {

```

Comment:
May I ask you to revert _all_ `final` additions.
They are not wrong but do not really belong to this PR..

Reply:
Done (it was due to my IDE configuration)